src/main/java/com/keeper/homepage/domain/election/application/convenience/ValidCandidateService.java

```diff
@@ -0,0 +1,59 @@
+package com.keeper.homepage.domain.election.application.convenience;
+
+import static com.keeper.homepage.global.error.ErrorCode.ELECTION_CANDIDATE_CANNOT_REGISTER;
+import static com.keeper.homepage.global.error.ErrorCode.ELECTION_CANDIDATE_NOT_FOUND;
+import static com.keeper.homepage.global.error.ErrorCode.MEMBER_JOB_NOT_FOUND;
+
+import com.keeper.homepage.domain.election.dao.ElectionCandidateRepository;
+import com.keeper.homepage.domain.election.entity.ElectionCandidate;
+import com.keeper.homepage.domain.member.dao.role.MemberJobRepository;
+import com.keeper.homepage.domain.member.entity.job.MemberJob;
+import com.keeper.homepage.global.error.BusinessException;
+import java.util.EnumSet;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class ValidCandidateService {
+
+  private static final long VIRTUAL_ELECTION_CANDIDATE_ID = 1;
+
+  private final ElectionCandidateRepository electionCandidateRepository;
+  private final MemberJobRepository memberJobRepository;
+
+  public ElectionCandidate findById(long candidateId) {
+    return electionCandidateRepository.findByIdAndIdNot(candidateId, VIRTUAL_ELECTION_CANDIDATE_ID)
+        .orElseThrow(() -> new BusinessException(candidateId, "candidateId", ELECTION_CANDIDATE_NOT_FOUND));
+  }
+
+  public MemberJob findJobById(long memberJobId) {
+    if (!isValidMemberJobId(memberJobId)) {
+      throw new BusinessException(memberJobId, "memJobId", ELECTION_CANDIDATE_CANNOT_REGISTER);
+    }
+    return memberJobRepository.findById(memberJobId)
+        .orElseThrow(() -> new BusinessException(memberJobId, "memberJobId", MEMBER_JOB_NOT_FOUND));
+  }
+
+  private boolean isValidMemberJobId(long memberJobId) {
+    return EnumSet.of(MemberJobId.ROLE_회장, MemberJobId.ROLE_부회장, MemberJobId.ROLE_총무).stream()
+        .anyMatch(jobId -> jobId.id == memberJobId);
```

Comment:
```suggestion
    return Arrays.stream(CandidateJob.values())
        .anyMatch(job -> job.id == memberJobId);
```

어떨까요? 그리고 `enum` 이름은 후보자 직위인 `CandidateJob`은 어떤가용?

Reply:
이 부분이 깔끔하지 못하다고 생각해서 저도 고민중이었는데, `Enum`을 따로 만들어두는게 괜찮을 지, 아니면 원래 List 방식에서 상수로 받는 것이 아닌` ROLE_회장.getId()` 형태로 받는게 좋을지 고민입니당..

Reply:
이 부분 수정 완료했습니다!
```java
  private static final List<Long> VALID_MEMBER_JOB_IDS = Arrays.asList(ROLE_회장.getId(), ROLE_부회장.getId(),
      ROLE_총무.getId());
```
이렇게 받는 게 더 깔끔해보여서 수정했습니다!

Reply:
아니면 기존에 있는 `MemberJob`에 메서드를 추가하는 것도 깔끔한 방법이 될 것 같아용!

```java
public boolean isCandidate() {
    return this.type == ROLE_회장 || this.type == ROLE_부회장 || this.type == ROLE_총무;
}
```